decisions/central server backup.md

```diff
@@ -0,0 +1,113 @@
+2. We also have Grafana installed on most central instances, and I think we've been using the same postgres instance for both omSupply central and Grafana
+3. Deal with or have a strategy to deal with misalignment of sync data between omSupply remote sites, omSupply central and mSupply central. Consider that omSupply or mSupply central data is restored, but there is some missalignment in recency, resulting in scenariors below
+
+|  scenario | mSupply Central (MC) | omSupply Central (OMC) | omSupply Remote (OMR) | Side effect                                                                                                                                                                                                                                                                                                                                                                                                                 |
+|:---------:|:--------------------:|:----------------------:|:---------------------:|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|   B.3.1   |          old         |         recent         |         recent        | Loss of data while restore of OMR data (via sync initialisation), due to missing data for that site on MC. OMC and OMR may miss some central data being added, since their cursor for that data would be ahead of central_change_log on MC. Some OMR remote records that are pushed to OMC may be related to central records on MC that are not present anymore, thus when OMR is re-initialised some constraints may fail. |
+|   B.3.2   |        recent        |           old          |         recent        | Loss of data while restore of OMR (via sync initialisation), due to missing data for that site on OMC. Some data being pushed OMC from OMR may not be integrated due to missing relation dependencies (that were pushed previously but are now missing). Some new OMC central data may not travel to OMR since the cursor of OMR for central data would be ahead of current OMC                                             |
+|   B.3.3   |        recent        |         recent         |          old          | This would require some restore of backup on omSupply remote site, which is not something we recommend or support (if remote site data is lost, you should re-initialise)                                                                                                                                                                                                                                                   |
```

Comment:
I have one key question - isn't B.3.3 essentially what we're trying to achieve? That's basically what we rely on for current MC prior to OMS2.0? But the complication I see is: don't we require both MC and OMC to backed up together at precisely the same time? Otherwise we're constantly going to end up with B.3.1 and B.3.2? At least for the cloud files anyway

Currently MC pauses all processes while it executes the backup process, including sync APIs. Would we not need to do the same with OMC at the same time?

Reply:
> Currently MC pauses all processes while it executes the backup process, including sync APIs. Would we not need to do the same with OMC at the same time?

Hmm i think to be fully consistent yeah, we would do this, and do both backups at exactly the same time. I added another Option 4 (which i already went with, oops), because it would deal with super pressing use case, where local data is either corrupted or completely lost, yes there would be inconsistencies between omSup central and mSup central, but it would be better then completely loosing all of omSup data (omSup will start to be populated starting in 2.2, with RnR forms. Also 2.2 backup also include files, which woudl be important if Asset functionality will start to be used)

I have one key question - isn't B.3.3 essentially what we're trying to achieve?

I actually think that most common use case would be when omSup would for some reason be corrupted or server would be lost. I think we already deal with B.3.3 with re-initialisation, and we shouldn't recommend backup on remote sites (beyond sync)

Reply:
Yea opt4 better than nothing! 

>I actually think that most common use case would be when omSup would for some reason be corrupted or server would be lost. I think we already deal with B.3.3 with re-initialisation, and we shouldn't recommend backup on remote sites (beyond sync)

Heh I've been trying to figure out what I meant in May... I am on board with remotes not having a backup other than reinit - I just meant we're gunning for ensuring we have recent backups of OM and OMC together and any option that doesn't achieve that doesn't pass the sniff test. The recency question then just boils down to how close together the OM and OMC backups are, with the ultimate goal of being lockstep with each other.